### DISCUSSION TITLE
After the second login (after install) the SSH password does not work anymore

### DISCUSSION THREAD

**Discussion by solvind-smartvind-machine-user** [MichaIng/DietPi#5557] [Q&A] [answered]
The following problem.

1. I have flashed DietPi to a MicroSD via RaspberryPi Imager (MacOS). I have already assigned SSH password in this tool.
2. During the initial setup in RaspberryPi I have set a global username and password, because it ask for it.
3. After the installtion I can login
4. When I later want to log in again via SSH, it does not accept the password anymore.

This happened repeatedly and could only be fixed by flashing the MicroSD again.

**Comment by Joulinar** [comment]
Some questions

Point 1: what kind of password you have assigned and where. Usually there is no need to assign a password while flashing the image

Point 2: what kind of user name you have set? Because there is no point during our installation where you could set a user. We just ask for a global software password and to set a password for user root and user dietpi

**Reply by solvind-smartvind-machine-user** [reply]
Point 1: please see screenshot [attached.] It's called OS modifiaction. 
Point 2; Yes, I'm talking about global software pw.
(url)
<img width="681" alt="Bildschirmfoto 2022-06-10 um 20 30 54" src="https://user-images.githubusercontent.com/78045213/173128640-e3f201ac-e69d-4c67-aa8c-348daa77b142.png">

**Reply by MichaIng** [reply]
This OS modification feature works with Raspberry Pi OS only, not with DietPi. It actually shouldn't be offered when selecting any other OS than the official RPi OS. This was showing up when you extracted and selected the DietPi `.img` file with the RPi imager?

Use `dietpi.txt` to change SSH password and such things on DietPi.

**Comment by Joulinar** [comment]
Ah you are using the RPi imager. Usually we don't have the user pi on our systems. We use root and dietpi. As well SSH is enabled by default.

**Reply by solvind-smartvind-machine-user** [reply]
I did not suggest the user Pi, that was just an example screenshot. I have created "root" and set a pw. 

Nevertheless I have set a global software pw during the initial setup. I could log in 2-3x with root and my pw via ssh.

**Reply by Joulinar** [reply]
As stated above. There is no need to set an password on that imager tool. Simply flash the image as is. If you like to set a password, use the configuration file `dietpi.txt` once the image has been flashed and before first boot.

**Reply by solvind-smartvind-machine-user** [reply]
I also tried to set the pw through dietpi.txt. It was the same result. First it worked and then at some point it didn't.

**Reply by Joulinar** [reply]
Did you tried using user dietpi?

**Reply by solvind-smartvind-machine-user** [reply]
Tried it right now. Permission denied. I tried my own pw and the general pw

**Reply by Joulinar** [reply]
Are there any configuration steps you have done in addition to the initial setup? Some software installation? Or changed any SSH server configuration?

**Reply by solvind-smartvind-machine-user** [reply]
I installed PiHole and owncloud. I mounted my extnernal drive to mnt/usb. Furthermore I configured DHCP to static IP. I was also surprised that it did not take my static IP from the dietpi.txt. I have the feeling that the changes are simply not taken over.

**Reply by Joulinar** [reply]
if changes are not taken over, you should have issues already after first reboot and not after some time/reboot

**Comment by MichaIng** [comment, marked as answer]
This OS modification feature works with Raspberry Pi OS only, not with DietPi. It actually shouldn't be offered when selecting any other OS than the official RPi OS. This was showing up when you extracted and selected the DietPi `.img` file with the RPi imager?

Use `dietpi.txt` to change SSH password and such things on DietPi.
